### ISSUE TITLE
Fn::GetAtt with S3 w/serverless-localstack

### ISSUE THREAD

**Issue by askharryclarke** [localstack/localstack#1549]
I have a simple S3 bucket 'MyBucket' declared alongside an IAM Role 'MyBucketRole', if I attempt to access the bucket's ARN via:
{ "Fn::GetAtt": ["MyBucket", "Arn"] }
it will throw:
Template error: resource MyBucket does not support attribute type Arn in Fn::GetAtt

However, if I attempt to access the ARN via:
{ "Fn::GetAtt": [{"Fn::GetAtt": ["MyBucket", "BucketName"]}, "Arn"] }
it works fine.

Under normal Serverless circumstance the first option works fine, am I being silly or is this a legitimate/known issue?

It also appears that the working version only works if the resource is already deployed. If attempting to launch both bucket and role attached to bucket at the same time, results in the following error:

ERROR:localstack.services.cloudformation.cloudformation_starter: Unable to parse and create resource "MyBucketRole": unhashable type: 'dict'

Serverless also throws a 500 error.

**The following log may also help**, 
This occurs when attempting to get the resource's ARN upon launching it in the same stack as the role attempting to reference it:

```
2019-09-09T13:22:54:ERROR:localstack.services.cloudformation.cloudformation_starter: Unable to parse and create resource "MyBucketRole": 400 Bad Request: <ErrorResponse xmlns="http://cloudformation.amazonaws.com/doc/2010-05-15/">
  <Error>
    <Type>Sender</Type>
    <Code>ValidationError</Code>
    <Message>Template error: resource MyBucket does not support attribute type Arn in Fn::GetAtt</Message>
  </Error>
  <RequestId>cf4c737e-5ae2-11e4-a7c9-ad44eEXAMPLE</RequestId>
</ErrorResponse>
```

**Comment by askharryclarke** [comment]
UPDATE: Seems as though it also happens with the exact same syntax and referencing a Kinesis Stream's ARN. I'm gonna make the assumption that this would happen with multiple resource types.

**Comment by abhyuditjain** [comment]
I am facing the same issue with localstack and serverless-localstack. Any update on this?

**Comment by TreeMan360** [comment]
An update on this would be great as it's currently a blocker for me

**Comment by whummer** [comment]
Thanks for reporting @askharryclarke @abhyuditjain @TreeMan360 - using `Fn::GetAtt` for `Arn` of S3 buckets should be fixed in #1879 . Please report here if the problem persists. Thanks
